core/src/main/java/org/infinispan/configuration/cache/IndexOverlay.java

```diff
@@ -22,21 +22,6 @@ void apply(TypedProperties properties) {
          properties.putIfAbsent(INDEX_MANAGER, "near-real-time");
          properties.putIfAbsent(READER_STRATEGY, "shared");
       }
-   },
-
-   /**
-    * Default for indexing a distributed cache. It will store indexes
-    * in Infinispan itself, with a master/slave backend in order to
-    * avoid holding cluster-wide locking during indexing
-    */
-   DISTRIBUTED_INFINISPAN {
```

Comment:
How about just deleting the IndexOverlay enum altogether and putting those autoconfigs directly in the builder as private constants ?

Reply:
sounds good

Reply:
done